src/VkBootstrap.cpp

```diff
@@ -1467,70 +1410,61 @@ detail::Result<Device> DeviceBuilder::build() const {
-	bool has_phys_dev_features_2 = false;
-	bool has_phys_dev_vulkan_features_11 = false;
-	bool has_phys_dev_vulkan_features_12 = false;
-	for (auto& pNext_struct : pNext_chain) {
-		if (pNext_struct->sType == VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_FEATURES_2) {
-			has_phys_dev_features_2 = true;
-		}
-		if (pNext_struct->sType == VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_VULKAN_1_1_FEATURES) {
-			has_phys_dev_vulkan_features_11 = true;
-		}
-		if (pNext_struct->sType == VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_VULKAN_1_2_FEATURES) {
-			has_phys_dev_vulkan_features_12 = true;
-		}
-	}
-
-// Add the correct structs to the pNext chain if api is 1.1/1.2 and aren't already present
-// This is to guard against users who were already doing that
 #if defined(VK_API_VERSION_1_1)
-	VkPhysicalDeviceFeatures2 local_features2{};
-	if (physical_device.instance_version >= VK_MAKE_VERSION(1, 1, 0) && !has_phys_dev_features_2) {
-		local_features2.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_FEATURES_2;
-		local_features2.features = physical_device.features;
-		pNext_chain.push_back(reinterpret_cast<VkBaseOutStructure*>(&local_features2));
-		has_phys_dev_features_2 = true;
-	}
-#if defined(VK_API_VERSION_1_2)
-	VkPhysicalDeviceVulkan11Features local_features_11 = physical_device.features_11;
-	VkPhysicalDeviceVulkan12Features local_features_12 = physical_device.features_12;
-	if (physical_device.instance_version >= VK_MAKE_VERSION(1, 2, 0)) {
-		if (!has_phys_dev_vulkan_features_11) {
-			pNext_chain.push_back(reinterpret_cast<VkBaseOutStructure*>(&local_features_11));
-			has_phys_dev_vulkan_features_11 = true;
-		}
-		if (!has_phys_dev_vulkan_features_12) {
-			pNext_chain.push_back(reinterpret_cast<VkBaseOutStructure*>(&local_features_12));
-			has_phys_dev_vulkan_features_12 = true;
+    VkPhysicalDeviceFeatures2 local_features2{};
+    local_features2.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_FEATURES_2;
+	auto physical_device_extension_features_copy = physical_device.extended_features_chain;
+    if (physical_device.instance_version >= VK_MAKE_VERSION(1, 1, 0)) {
+        has_phys_dev_features_2 = true;
+		for(auto& features_node : physical_device_extension_features_copy) {
+			final_pnext_chain.push_back(reinterpret_cast<VkBaseOutStructure*>(&features_node));
 		}
 	}
-#endif
 #endif
 
 	VkDeviceCreateInfo device_create_info = {};
 	device_create_info.sType = VK_STRUCTURE_TYPE_DEVICE_CREATE_INFO;
-	detail::setup_pNext_chain(device_create_info, pNext_chain);
 	device_create_info.flags = info.flags;
 	device_create_info.queueCreateInfoCount = static_cast<uint32_t>(queueCreateInfos.size());
 	device_create_info.pQueueCreateInfos = queueCreateInfos.data();
 	device_create_info.enabledExtensionCount = static_cast<uint32_t>(extensions.size());
 	device_create_info.ppEnabledExtensionNames = extensions.data();
+
+	for(auto& pnext : info.pNext_chain) {
+        final_pnext_chain.push_back(pnext);
+	}
+
+	detail::setup_pNext_chain(device_create_info, final_pnext_chain);
```

Comment:
Do the pNext chain setup after the local_features2 struct has been added. Makes the logic easier.